packages/sitecore-jss/src/graphql/site-query-service.ts

```diff
@@ -0,0 +1,107 @@
+import { DocumentNode } from 'graphql';
+import { BaseQueryService, BaseQueryVariables } from './base-query-service';
+// TODO: rewrite site query results
+// TODO: review-rewrite comments
+
+/**
+ * Schema of data returned in response to a "site" query request
+ * @template T The type of objects being requested.
+ */
+export interface SiteQueryResult<T> {
```

Comment:
This is done in line with implementation in search-service. The approach could be better, but it also makes sense in its current form: there's a set format returned by the service - but the results needed by consumers of the class may differ. 
In case of sitemap service - we only need paths.
In case of personalize task (that is in progress) - we would also need segments. Both returned by site query - but the end result format is taken outside of the site-service.

Reply:
I do agree this could be done better, but we probably need to do it in part of an overall refactor around graphql to bring it to a single standard.

Reply:
Got it. I guess part of my point is that using the name prefix `SiteQuery` is a bit misleading. The `site` query includes much more than just routes and these are very much so related to routes. I suppose just a rename of these to reflect that would help (e.g. `SiteRoutes`). However, search-service was reused among multiple classes (previous `GraphQLSitemapService` and `GraphQLDictionaryService`) whereas I'm not sure these would be? I know you mentioned personalize, but I thought we had decided to bake that directly into the `GraphQLSitemapService` (no need for a separate class).

Reply:
Moved the logic into GraphQLSitemapService now - and did some cleanup related to this change, please take a look